File: backend/health-platform/src/main/java/com/healthplatform/healthplatform/hospital/entity/Hospital.java
package com.healthplatform.healthplatform.hospital.entity;

import com.healthplatform.healthplatform.common.entity.BaseEntity;
import com.healthplatform.healthplatform.user.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "hospitals")
@Getter
@Setter
@NoArgsConstructor
public class Hospital extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(
            name = "user_id",
            nullable = false
    )
    private User user;

    @Column(
            name = "name",
            nullable = false,
            length = 200
    )
    private String name;

    @Column(
            name = "city",
            length = 100
    )
    private String city;

    @Column(
            name = "address",
            columnDefinition = "TEXT"
    )
    private String address;

    @Column(
            name = "phone",
            length = 50
    )
    private String phone;

    @Column(
            name = "notes",
            columnDefinition = "TEXT"
    )
    private String notes;
}

File: frontend/src/types/hospital.ts
export type Hospital = {
  id: string;
  name: string;
  city: string | null;
  address: string | null;
  phone: string | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
};

export type CreateHospitalRequest = {
  name: string;
  city: string | null;
  address: string | null;
  phone: string | null;
  notes: string | null;
};

export type UpdateHospitalRequest = {
  name: string;
  city: string | null;
  address: string | null;
  phone: string | null;
  notes: string | null;
};
